backend/src/main/java/org/oaebudt/edc/backend/controller/BaseController.java

```diff
@@ -0,0 +1,20 @@
+package org.oaebudt.edc.backend.controller;
+
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/api")
+public class BaseController {
+
+    @GetMapping("/secure")
+    public String testSecureApi(){
```

Comment:
using `test` resources in production code makes me think that this should in fact be test code, and currently we're missing tests.

please add test coverage that shows how endpoints under `/api/public` have no authentication and all the others have it instead.

(additional note: is the "public" endpoint thing actually needed? I could had missed some pieces :') )